### PR TITLE
ADD Check all the context responses looking for an error

### LINKED ISSUE #153: Deal with multiple ContextResponses
When the contextBroker returns a response that is being served by different Context Providers, there may be multiple Context Responses with different error codes. The IoT Agent should cope with the different codes.

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #153

### CODE REVIEW

**Comment by XavierVal** [comment]
LGTM

**Comment by AlvaroVega** [comment]
LGTM

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1 +1,2 @@
-- ADD Command attributes to the initial entity
\ No newline at end of file
+- ADD Command attributes to the initial entity
+- ADD Checking for multiple context responses (#153)
```
Comment (gtorodelvalle) [inline CHANGES_NEXT_RELEASE]: Final blank line appreciated :)

File: lib/services/ngsiParser.js [hunk 2]
```diff
@@ -211,10 +211,13 @@ function getErrorField(body) {
     var errorField = body.errorCode ||
         body.orionError;
 
-    if (body && body.contextResponses &&
-        body.contextResponses[0] && body.contextResponses[0] &&
-        body.contextResponses[0].statusCode && body.contextResponses[0].statusCode.code !== '200') {
-        errorField = body.contextResponses[0].statusCode;
+    if (body && body.contextResponses) {
+        for (var i in body.contextResponses) {
+            if (body.contextResponses[i] && body.contextResponses[i] &&
```
Comment (gtorodelvalle) [inline lib/services/ngsiParser.js]: Love this double check... :p

File: test/unit/contextResponses/queryContext3Error.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      }
+    },
+    {
+      "statusCode": {
+        "code": "404",
+        "reasonPhrase": "Not Found"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/queryContext3Error.json]: Final blank line appreciated :)

File: lib/services/ngsiParser.js (hunk 2 above)
Comment (dmoranj) [inline lib/services/ngsiParser.js]: What do you mean by double?

Comment (dmoranj) [inline lib/services/ngsiParser.js]: Fixing. I'm even going to remove both, as that check is implicit in the foreach.

File: lib/services/ngsiParser.js [hunk 4]
```diff
@@ -211,10 +211,12 @@ function getErrorField(body) {
     var errorField = body.errorCode ||
         body.orionError;
 
-    if (body && body.contextResponses &&
-        body.contextResponses[0] && body.contextResponses[0] &&
-        body.contextResponses[0].statusCode && body.contextResponses[0].statusCode.code !== '200') {
-        errorField = body.contextResponses[0].statusCode;
+    if (body && body.contextResponses) {
+        for (var i in body.contextResponses) {
+            if (body.contextResponses[i].statusCode && body.contextResponses[i].statusCode.code !== '200') {
```
Comment (gtorodelvalle) [inline lib/services/ngsiParser.js]: Ups, sorry! Maybe I confused you :) I would suggest:
```javascript
if (body.contextResponses[i] && body.contextResponses[i].statusCode && body.contextResponses[i].statusCode.code !== '200') {
```
(notice the `body.contextResponses[i]` check) to protect the code against the case of an empty contextResponses array :)

**Comment by gtorodelvalle** [comment]
LGTM and merging :) Thanks @dmoranj !

**Pull request merged**